### DISCUSSION TITLE
consider `snmalloc` as an alternative to `mimalloc`

### DISCUSSION THREAD

**Discussion by enihcam** [rui314/mold#1104] [Ideas]
Since `mimalloc` has been unmaintained for a long while, I think it's probably time to add a build option to allow users choosing alternative memory allocator, e.g. [snmalloc](https://github.com/microsoft/snmalloc).

@rui314

**Comment by rui314** [comment]
Is mimalloc really unmaintained? The last release I can find at https://github.com/microsoft/mimalloc/tags is v2.1.2 which was released in April this year.

**Reply by enihcam** [reply]
there's no activities taken place after July 1st.

https://github.com/microsoft/mimalloc/activity

**Reply by rui314** [reply]
Three months are too short to consider the library unmaintained.

**Reply by Sir-Walrus** [reply]
https://github.com/microsoft/mimalloc/graphs/code-frequency says wildly variable commit frequency is normal for this repo. https://github.com/microsoft/snmalloc/issues/273#issuecomment-1720708560 also says there's nothing to worry about.

**Comment by X547** [comment]
@rui314, What is a problem with mimalloc on 32 bit systems?

https://github.com/rui314/mold/blob/main/CMakeLists.txt#L179

**Comment by rui314** [comment]
I don't remember that now, but that should be reproducible if you try to run mold with mimalloc.
